File: Java/src/Taller/Taller3/DiscountRechargeableSubwayCard.java
package Taller.Taller3;

public class DiscountRechargeableSubwayCard extends RechargeableSubwayCard{
    private double discountPercentage;
    public DiscountRechargeableSubwayCard(SubwayCentral sc, double discountPercentage){
        super(sc);
        this.discountPercentage=discountPercentage;
    }

    @Override
    protected double rideCost() {
        return super.rideCost()*((100-discountPercentage)/100);
    }
}

File: Ruby/Guias/G7/Ej6/discount_rechargeable_subway_card.rb
require_relative 'rechargeable_subway_card'
class DiscountRechargeableSubwayCard < RechargeableSubwayCard
  def initialize(central,discount)
    raise "Not a central" unless central.is_a?(SubwayCentral)
    raise "Not a numeric discount" unless discount.is_a?(Numeric) && discount >= 0 && discount <= 1
    super(central)
    @discount = discount
  end

  def ride_cost
    super * (1-@discount)
  end
end